test/API/H5_api_link_test.c

```diff
@@ -2151,7 +2162,7 @@ test_create_external_link_dangling(void)
         TEST_ERROR;
     if (remove_test_file(NULL, ext_link_filename) < 0)
         TEST_ERROR;
-
+    free(ext_link_filename);
```

Comment:
Fine for these cases based on placement, but for the future it's always good practice to set these pointers back to NULL after a `free` call. Otherwise, if someone adds something else in the future past this point that can fail, you may end up double-freeing `ext_link_filename`, for example.